lib/readStreamComponents.js

```diff
@@ -89,27 +92,51 @@ function createNameConverter() {
 var createDocumentGenerator = function(kunnat, maakunnat) {
   return map_stream.obj(function(record) {
 
-    var mmlDoc = new Document(record.type, "mml-" + record["pnr:paikkaID"]);
-
-    var phrases = {};
+    var mmlDoc = new Document("nlsfi", 'venue', "mml-" + record["pnr:paikkaID"]);
     record["pnr:nimi"].forEach(function(name) {
       mmlDoc.setName(name.lang, name.name);
-      phrases[name.lang] = name.name;
     });
 
-    mmlDoc.phrase = phrases;
+    var kuntaKoodi = record["pnr:kuntaKoodi"];
+    var maakuntaKoodi = record["pnr:maakuntaKoodi"];
+    var kunta = kunnat[kuntaKoodi];
+    var maakunta = maakunnat[maakuntaKoodi];
+    var maakuntaNimi = maakunta[0].text;
+    var kuntaNimi = kunta[0].text;
+
+    mmlDoc.setAdmin('locality', kuntaNimi);
+    mmlDoc.setAdmin('admin1', maakuntaNimi);
+    mmlDoc.setAdmin('admin0', 'Finland');    mmlDoc.setAlpha3('FIN');
 
-    mmlDoc.setAdmin('locality', kunnat[record["pnr:kuntaKoodi"]][0].text);
-    mmlDoc.setAdmin('admin1', maakunnat[record["pnr:maakuntaKoodi"]][0].text);
+    if(record.type) {
+      mmlDoc.addCategory(record.type);
+    }
+
+    kuntaKoodi = "mml-" + kuntaKoodi;
+    maakuntaKoodi = "mml-" + maakuntaKoodi;
 
-    mmlDoc.setAdmin('admin0', 'Finland');
+    for(var i in kunta) {
+      try { mmlDoc.addParent('locality', kunta[i].text, kuntaKoodi); }
+      catch (err) { logger.info('invalid locality', err) }
+    }
+
+    for(var i in maakunta) {
+      try { mmlDoc.addParent('region', maakunta[i].text, maakuntaKoodi); }
```

Comment:
You could try to get the correct wof id for the region and municipality, but we'll go with these for now.